homeassistant/components/opensprinkler/config_flow.py

```diff
@@ -0,0 +1,85 @@
+"""Config flow for Opensprinkler integration."""
+import logging
+
+from pyopensprinkler import (
+    OpenSprinkler,
+    OpensprinklerAuthError,
+    OpensprinklerConnectionError,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries, exceptions
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PASSWORD, CONF_PORT
+
+from .const import (  # pylint: disable=unused-import
+    CONF_RUN_SECONDS,
+    DEFAULT_NAME,
+    DEFAULT_PORT,
+    DEFAULT_RUN_SECONDS,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEVICE_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_PASSWORD): str,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): int,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): str,
+        vol.Optional(CONF_RUN_SECONDS, default=DEFAULT_RUN_SECONDS): int,
```

Comment:
Let's drop this from device schema and allow configuring this later via an Option Flow.